flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AwsV2Util.java

```diff
@@ -66,254 +36,23 @@
-
-    /**
-     * Creates an Amazon Kinesis Async Client from the provided properties. Configuration is copied
-     * from AWS SDK v1 configuration class as per: -
-     * https://github.com/aws/aws-sdk-java-v2/blob/2.13.52/docs/LaunchChangelog.md#134-client-override-retry-configuration
-     *
-     * @param configProps configuration properties
-     * @param clientConfiguration the AWS SDK v1.X config ported to V2 to instantiate the client
-     * @param httpClient the underlying HTTP client used to talk to Kinesis
-     * @return a new Amazon Kinesis Client
-     */
-    public static KinesisAsyncClient createKinesisAsyncClient(
-            final Properties configProps,
-            final ClientConfiguration clientConfiguration,
-            final SdkAsyncHttpClient httpClient) {
-        final ClientOverrideConfiguration overrideConfiguration =
-                createClientOverrideConfiguration(
-                        clientConfiguration, ClientOverrideConfiguration.builder());
-        final KinesisAsyncClientBuilder clientBuilder = KinesisAsyncClient.builder();
-
-        return createKinesisAsyncClient(
-                configProps, clientBuilder, httpClient, overrideConfiguration);
-    }
-
-    public static SdkAsyncHttpClient createHttpClient(
-            final ClientConfiguration config,
-            final NettyNioAsyncHttpClient.Builder httpClientBuilder,
-            final Properties consumerConfig) {
-
-        int maxConcurrency =
-                Optional.ofNullable(consumerConfig.getProperty(EFO_HTTP_CLIENT_MAX_CONCURRENCY))
-                        .map(Integer::parseInt)
-                        .orElse(DEFAULT_EFO_HTTP_CLIENT_MAX_CONCURRENCY);
-
-        Duration readTimeout =
-                Optional.ofNullable(consumerConfig.getProperty(EFO_HTTP_CLIENT_READ_TIMEOUT_MILLIS))
-                        .map(Integer::parseInt)
-                        .map(Duration::ofMillis)
-                        .orElse(DEFAULT_EFO_HTTP_CLIENT_READ_TIMEOUT);
-
-        httpClientBuilder
-                .maxConcurrency(maxConcurrency)
-                .connectionTimeout(Duration.ofMillis(config.getConnectionTimeout()))
-                .readTimeout(readTimeout)
-                .tcpKeepAlive(config.useTcpKeepAlive())
-                .writeTimeout(Duration.ofMillis(config.getSocketTimeout()))
-                .connectionMaxIdleTime(Duration.ofMillis(config.getConnectionMaxIdleMillis()))
-                .useIdleConnectionReaper(config.useReaper())
-                .protocol(Protocol.HTTP2)
-                .connectionAcquisitionTimeout(CONNECTION_ACQUISITION_TIMEOUT)
-                .http2Configuration(
-                        Http2Configuration.builder()
-                                .healthCheckPingPeriod(HEALTH_CHECK_PING_PERIOD)
-                                .initialWindowSize(INITIAL_WINDOW_SIZE_BYTES)
-                                .build());
```

Comment:
I cannot see where these configs have been moved to? I can see `createKinesisAsyncClient` in `AWSKinesisDataStreamsUtil.java`, however it sets a subset of config

Reply:
In `AWSKinesisDataStreamsUtil.java`, the missing assignments are done in `NettyNioAsyncHttpClient.Builder::buildWithDefaults` when extracting the properties from the provided `AttributeMap`.